Проектная работа №1

Barbossa1 opened the pull request:
"24 часа Ле-Мана" v0.1

Barbossa1 committed:
"24 часа Ле-Мана" v0.1

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,140 @@
+        // Ввод третьего автомобиля
+        String thirdCarName = getCorrectCarName(scanner);
+        int thirdCarSpeed = getCorrectCarSpeed(scanner);
+        createCar(thirdCarName, thirdCarSpeed, carsList);
```
Эти три блока кода повторяются, их можно вынести в отдельную функцию или в данном случае удобнее воспользоваться циклами

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,140 @@
+
+        // Гонка
+        Race race = new Race();
+        for (Car car : carsList) {
```
От хранения списка машин и лишнего цикла при определении победителя можно избавиться, если при вводе данных сразу вычислять победителя и хранить его в отдельной переменной, тогда программа будет требовать меньше памяти и работать быстрее

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,140 @@
+            try {
+                speed = Integer.parseInt(input);
+
+                if (speed > 0 && speed < 250) {
```
Константы для максимальной и минимальной скорости лучше вынести в константы для повышения читабельности кода

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,140 @@
+    }
+
+    // Класс «Автомобиль»
+    public static class Car {
```
Класс не стоит делать статическим, у него есть конструктор. В данном случае необходимо писать `static`, так как класс объявлен в другом классе - `Main`, но от такой вложенности нет смысла - два класса никак не связаны друг с другом, поэтому объявление этого класса можно вынести на верхний уровень или в отдельный файл

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,140 @@
+    // Класс «Автомобиль»
+    public static class Car {
+        String carName;
+        int carSpeed;
```
Поля лучше пометить `final`, тем самым исключив возможность их модификации извне

Barbossa1 committed:
1. Весь код не написан в одном классе Main.
2. Повторяющийся блок заменен циклом.
3. При вводе данных сразу вычисляется победитель.
4. Константы для максимальной и минимальной скорости.
5. Константы для полей CAR_NAME и CAR_SPEED.
